example/memcached-operator/controller/memcached.go

```diff
@@ -0,0 +1,130 @@
+package controller
+
+import (
+	examplev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/example/memcached-operator/controller/resource"
+	"github.com/giantswarm/operatorkit/example/memcached-operator/logger"
+	"github.com/giantswarm/operatorkit/informer"
+)
+
+const name = "memcached-operator"
+
+// Config represents the configuration used to create a new memcached controller.
+type Config struct {
+	K8sClient    kubernetes.Interface
+	K8sExtClient apiextensionsclient.Interface
+	G8sClient    versioned.Interface
+}
+
+// Memcached is a type containing the OperatorKit controller.
+type Memcached struct {
+	*controller.Controller
+}
+
+// New creates a new memcached controller.
+func NewMemcached(config Config) (*Memcached, error) {
+	var err error
+
+	var (
+		crd        = examplev1alpha1.NewMemcachedConfigCRD()
+		restClient = config.G8sClient.ExampleV1alpha1().RESTClient()
+		watcher    = config.G8sClient.ExampleV1alpha1().MemcachedConfigs("")
+	)
+
+	var deploymentsResource controller.Resource
+	{
+		c := resource.DeploymentsConfig{}
+
+		deploymentsResource, err = resource.NewDeployments(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var servicesResource controller.Resource
+	{
+		c := resource.ServicesConfig{}
+
+		servicesResource, err = resource.NewServices(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resources := []controller.Resource{
+		deploymentsResource,
+		servicesResource,
+	}
+
+	// Below is a common controller wiring. This code doesn't change unless
+	// you want to reconcile non-CRD objects or to use more sophisticated
+	// object routing.
```

Comment:
This comment about it being the wiring is really useful. But I think it would be good to have comments for the informer and the controller anyway. So it's clear what they do, for people reading the example.

Reply:
Can you please provide those comments in separate PR? I think it will be quicker that way.

Reply:
Good plan. I'll do that.